daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -1759,23 +1759,52 @@ class FPGAReshape(ONNXForward):
     def forward(node: ONNXOp, state: SDFGState,
                 sdfg: SDFG) -> typing.Union[Node, SDFG]:
         node.validate(sdfg, state)
-        if (in_desc_with_name(node, state, sdfg, "data").dtype !=
-                out_desc_with_name(node, state, sdfg, "reshaped")):
+
+        input_name = "data"
+        output_name = "reshaped"
+        flatten = False
+
+        # if called from Flatten
+        if "input" in node._in_connectors.keys():
+            input_name = "input"
+            output_name = "output"
+            flatten = True
+
+        if (in_desc_with_name(node, state, sdfg, input_name).dtype !=
+                out_desc_with_name(node, state, sdfg, output_name).dtype):
```

Comment:
This will already be checked by `node.validate()` before the expansion is called